plugin/trino-exasol/src/test/java/io/trino/plugin/exasol/TestExasolTypeMapping.java

```diff
@@ -0,0 +1,261 @@
+                .execute(getQueryRunner(), exasolCreateAndInsert(qualified("test_decimal")));
+    }
+
+    @Test
+    public void testChar()
+    {
+        SqlDataTypeTest.create()
+                .addRoundTrip("char(10)", "NULL", createCharType(10), "CAST(NULL AS char(10))")
+                .addRoundTrip("char(6)", "'text_a'", createCharType(6), "CAST('text_a' AS char(6))")
+                .addRoundTrip("char(5)", "'攻殻機動隊'", createCharType(5), "CAST('攻殻機動隊' AS char(5))")
+                .addRoundTrip("char(1)", "'😂'", createCharType(1), "CAST('😂' AS char(1))")
+                .execute(getQueryRunner(), exasolCreateAndInsert(qualified("test_char")));
+    }
+
+    private String qualified(String tableName)
+    {
+        return TEST_SCHEMA + "." + tableName;
+    }
+
+    @Test
+    public void testVarchar()
+    {
+        SqlDataTypeTest.create()
+                .addRoundTrip("varchar(10)", "NULL", createVarcharType(10), "CAST(NULL AS varchar(10))")
+                .addRoundTrip("varchar(10)", "'text_a'", createVarcharType(10), "CAST('text_a' AS varchar(10))")
+                .addRoundTrip("varchar(255)", "'text_b'", createVarcharType(255), "CAST('text_b' AS varchar(255))")
+                .addRoundTrip("varchar(65535)", "'text_d'", createVarcharType(65535), "CAST('text_d' AS varchar(65535))")
+                .addRoundTrip("varchar(5)", "'攻殻機動隊'", createVarcharType(5), "CAST('攻殻機動隊' AS varchar(5))")
+                .addRoundTrip("varchar(32)", "'攻殻機動隊'", createVarcharType(32), "CAST('攻殻機動隊' AS varchar(32))")
+                .addRoundTrip("varchar(20000)", "'攻殻機動隊'", createVarcharType(20000), "CAST('攻殻機動隊' AS varchar(20000))")
+                .addRoundTrip("varchar(1)", "'😂'", createVarcharType(1), "CAST('😂' AS varchar(1))")
+                .addRoundTrip("varchar(77)", "'Ну, погоди!'", createVarcharType(77), "CAST('Ну, погоди!' AS varchar(77))")
+                .addRoundTrip("varchar(2000000)", "'text_f'", createVarcharType(2000000), "CAST('text_f' AS varchar(2000000))") // too long for a char in Trino
+                .execute(getQueryRunner(), exasolCreateAndInsert(TEST_SCHEMA + "." + "test_varchar"));
+    }
+
+    @Test
+    public void testUnsupportedType()
+    {
+        JdbcSqlExecutor jdbcSqlExecutor = exasolServer.getSqlExecutor();
+        String table = qualified("test_forced_varchar_mapping");
+        jdbcSqlExecutor.execute("CREATE TABLE " + table + "(geo_col geometry)");
+        jdbcSqlExecutor.execute("INSERT INTO " + table + "(geo_col) " + "VALUES ('POINT(1 2)')");
+        try {
+            assertQueryFails("SELECT 1 FROM " + table + " WHERE geo_col = 'POINT(1 2)'", "Table '.*' has no supported columns.*");
+        }
+        finally {
+            jdbcSqlExecutor.execute("DROP TABLE " + table);
+        }
+    }
+
+    @ParameterizedTest
+    @MethodSource("sessionZonesDataProvider")
```

Comment:
We're avoiding `@ParameterizedTest` annotation. See other type mapping test classes.